File: 205-isomorphic-strings/isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character, Character> ak = new HashMap<>();
        HashMap<Character, Character> kb = new HashMap<>();
        if(s.length() != t.length()) return false;
        int i = 0;
        int j = 0;
        while(i < s.length()){
            if(ak.containsKey(s.charAt(i))){
                if(ak.get(s.charAt(i)) != t.charAt(i)){
                    return false;
                }
            }
            else {
                ak.put(s.charAt(i), t.charAt(i));
            }
             if(kb.containsKey(t.charAt(i))){
                if(kb.get(t.charAt(i)) != s.charAt(i)){
                    return false;
                }
            }
            else {
                kb.put(t.charAt(i), s.charAt(i));
            }
            i++;
        }return true;
    }
}

File: 205-isomorphic-strings/isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        map<char, char> ak;
        map<char, char> kb;
        if(s.size() != t.size()) return false;
        for(int i = 0; i<s.size(); i++){
            char m = s[i];
            char j = t[i];
            if(ak.find(m) != ak.end()){
                if(ak[m] != j) return false;
            }
            else{
                ak[m] = j;
            }

            if(kb.find(j) != kb.end()){
                if(kb[j] != m) return false;
            }
            else{
                kb[j] = m;
            }
        }return true;
    }
};